armsrc/thinfilm.c

```diff
@@ -114,8 +114,10 @@ static int EmSendCmdThinfilmRaw(const uint8_t *resp, uint16_t respLen) {
     }
 
     // Ensure that the FPGA Delay Queue is empty
-    uint8_t fpga_queued_bits = FpgaSendQueueDelay >> 3;
-    for (i = 0; i <= (fpga_queued_bits >> 3) + 1;) {
+    uint16_t fpga_queued_bits = FpgaSendQueueDelay >> 3;
```

Comment:
explicitly use unsigned values for the maths, and store back as `uint16_t`.  This allows the `for()` loop to index on a variable of size `uint16_t`.

Reply:
good catch